theseus/src/state/projects.rs

```diff
@@ -140,23 +139,33 @@ async fn read_icon_from_file(
 ) -> crate::Result<Option<PathBuf>> {
     if let Some(icon_path) = icon_path {
         // we have to repoen the zip twice here :(
-        let zip_file = File::open(path)?;
-        if let Ok(mut zip) = ZipArchive::new(zip_file) {
-            if let Ok(mut file) = zip.by_name(&icon_path) {
-                let mut bytes = Vec::new();
-                if file.read_to_end(&mut bytes).is_ok() {
-                    let bytes = bytes::Bytes::from(bytes);
-
-                    let permit = io_semaphore.acquire().await?;
-                    let path = write_cached_icon(
-                        &icon_path, cache_dir, bytes, &permit,
-                    )
-                    .await?;
+        let zip_file_reader = ZipFileReader::new(path).await?;
```

Comment:
this shouldn't be "question-marked". if the zip is invalid the func shouldn't fail